README.md

```diff
@@ -4,45 +4,120 @@ The goal is to improve payment confirmation experience with the help of WebAuthn
 
 ![Screenshot](https://raw.githubusercontent.com/rsolomakhin/secure-payment-confirmation/master/payment.png)
 
+The rest of the document is organized into these sections:
+
+- [Creating a credential](#creating-a-credential)
+- [Querying a credential](#querying-a-credential)
+- [Authenticating a payment](#authentication-a-payment)
+
 ## Creating a credential
 
-A new credential type is introduced for `navigator.credentials.create()` that stores the name and the icons of a payment instrument.
+A new `SecurePaymentCredential` credential type is introduced for `navigator.credentials.create()` to bind descriptions of a payment instrument, i.e. a name and an icon, with a vanilla [PublicKeyCredential].
+
+Proposed new spec that extends [Web Authentication]:
+```webidl
+[SecureContext, Exposed=Window]
+interface SecurePaymentCredential : PublicKeyCredential {
+};
 
+partial dictionary CredentialCreationOptions {
+  SecurePaymentCredentialCreationOptions securePayment;
+};
+
+dictionary SecurePaymentCredentialCreationOptions {
+  required PublicKeyCredentialRpEntity rp;
+  required SecurePaymentCredentialInstrument instrument;
+  required BufferSource challenge;
+  required sequece<PublicKeyCredentialParameters> pubKeyCredParams;
+  unsigned long timeout;
+  
+  // PublicKeyCredentialCreationOption attributes that are intentionall omitted:
+  // user: For a SecurePaymentCredential, |instrument| is analogous to |user|.
+  // excludeCredentials: No payment use case has been proposed for this field.
+  // attestation: Authenticator attestation is considered an anti-pattern for adoption so will not be supported.
+  // extensions: No payment use case has been proposed for this field.
+};
+
+dictionary SecurePaymentCredentialInstrument {
+  required DOMString instrumentId;
+  required DOMString displayName;
+  required USVString icon;
```

Comment:
one icon might not be sufficient... the icon might get displayed in multiple contexts - like in a payment sheet or in a keyboard. In a keyboard, it would be maybe 32x32 pixels, so will need special treatment (or an entirely new icon) to show properly when compared to the one shown in a sheet (or even in a minimal UI, which may be say 100x100px)

Reply:
We'd like to run this as an origin trial on short timeframe, so let's go with something that's easiest to implement: a single URL. Origin trials have explicit agreement that the user agent is free to break the API after the origin trial expires, so we can enable usage of multiple icons in the future. Also, I believe there's some precedent in WebAuthn for using a single image URL.